src/Assert.h

```diff
@@ -4,4 +4,4 @@
 #include <stdlib.h>
 #include "Exception.h"
 #define PLATFORM_VERIFY(expression) do{ if((expression) == false) throw core::Exception(SOURCE, "An error occured, Reason - %s", strerror(errno)); } while(0)
-#define VERIFY(expression, format, ...) do{ if((expression) == false) throw core::Exception(SOURCE, format, ##__VA_ARGS__ ); } while(0)
+#define VERIFY(expression, ...) do{ if((expression) == false) throw core::Exception(SOURCE, __VA_ARGS__ ); } while(0)
```

Comment:
If we still want to keep the format, we can suppress the warning in GCC:
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wvariadic-macros"
#define FOO(A, ...) foo(A, ##__VA_ARGS__)
#pragma GCC diagnostic pop